src/__tests__/dataloader.test.js

```diff
@@ -695,33 +697,34 @@ describe('Accepts options', () => {
     expect(cacheMap.get('X')).toBe(promiseX);
   });
 
-  it('Complex cache behavior via clearAll()', async () => {
-    // This loader clears its cache as soon as a batch function is dispatched.
-    const loadCalls = [];
-    const identityLoader = new DataLoader<string, string>(keys => {
-      identityLoader.clearAll();
-      loadCalls.push(keys);
-      return Promise.resolve(keys);
-    });
-
-    const values1 = await Promise.all([
-      identityLoader.load('A'),
-      identityLoader.load('B'),
-      identityLoader.load('A'),
-    ]);
-
-    expect(values1).toEqual([ 'A', 'B', 'A' ]);
-
-    const values2 = await Promise.all([
-      identityLoader.load('A'),
-      identityLoader.load('B'),
-      identityLoader.load('A'),
-    ]);
-
-    expect(values2).toEqual([ 'A', 'B', 'A' ]);
-
-    expect(loadCalls).toEqual([ [ 'A', 'B' ], [ 'A', 'B' ] ]);
-  });
+  // In faster-dataloader, the cache doesn't get populated until the promise resolves
```

Comment:
Interpreting this comment, does this mean that:

1. suppose you call `loader.load(key)`. it takes 50ms to resolve the promise
2. 1ms after your first call to `loader.load(key)`, some other code path calls `loader.load(key)`

With the faster-dataloader, we'd make 2 calls to the underlying batch function. With the regular dataloader, we'd only make one call.

Is that right?

Reply:
Yup that's right. But, with https://github.com/ThreadsHQ/Web/pull/7471, I ended up going with the other PR where this case should now work properly. I'll take another pass on verifying this, making sure either this test case passes or there is some other test case that passes with the new code.